Avoid converting input example (#12782)
mlflow/mlflow: docs/source/model/signatures.rst

    It is **highly recommended** to always include an input example along with your models when you log them.

.. note::

>     In latest MLflow, we do not convert dictionary input example to pandas DataFrame anymore. If the model input is json-serializable object, e.g. list or dictionary,

Reviewer: ```suggestion
    Since ..., we do not convert dictionary input example to pandas DataFrame anymore. If the model input is json-serializable object, e.g. list or dictionary,
```

Can we use `since ...` instead to clarify when it starts.

>     In latest MLflow, we do not convert dictionary input example to pandas DataFrame anymore. If the model input is json-serializable object, e.g. list or dictionary,

Reviewer: I'd avoid using `we` because it's unclear what that refers to. Does that refer to MLflow devs? or Databricks employees? or someone else?

>     Since MLflow 2.16.0, dictionary input example is not converted to pandas DataFrame anymore. If the model input is json-serializable object, e.g. list or dictionary,
>     it is saved as json object without any conversion. For numpy arrays, a json serializable format is saved. For pandas DataFrame, it is converted to dictionary format 

Reviewer 2: ```suggestion
    Prior to MLflow 2.16.0, dictionary input example was converted to Pandas DataFrame format when saving. In later versions, the input example is simply saved in its JSON serialized format. For pandas DataFrame, it is converted to dictionary format 
```

nit: to make the nuance of the change a bit more clear

    params = {"temperature": 0.5, "top_k": 1}
    input_example = (input_data, params)
    mlflow.transformers.log_model(..., input_example=input_example)

> Model Serving Example
> ---------------------

Reviewer 2: ```suggestion
Model Serving Payload Example
-----------------------------
```

nit: The original one may sound we demonstrate how to serve models

> When logging a model with input example, a corresponding serving payload is automatically saved in ``serving_input_payload.json`` file,
> which is validated against the logged model prior to model deployment. The serving payload is
> converted from input example, and it is a json string that can be used when querying a deployed model endpoint. 

Reviewer 2: ```suggestion
Once an MLflow model is deployed to an REST endpoint for inference, the request payload will be a JSON serialized and may have subtle difference from in-memory representation. To validate your model works for inference, you can use the ``serving_input_payload.json`` file. It is automatically logged along with the model when an ``input_example`` is provided and contains a json format of the given input example for querying a deployed model endpoint.
```

nit: I would add some context for what is "serving payload" means and why we need a special validation for it.

> The following example demonstrates how to use the serving payload:

Reviewer 2: ```suggestion
The following example demonstrates how to load the serving payload from a logged model
```

.. code-block:: python

    import mlflow
    from mlflow.models.utils import load_serving_example_from_uri

    input_example = {
        "messages": [
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "assistant", "content": "What would you like to ask?"},
            {"role": "user", "content": "Who owns MLflow?"},
        ]
    }
    model_info = mlflow.langchain.log_model(..., input_example=input_example)
    print(f"model_uri: {model_info.model_uri}")
    serving_example = load_serving_example_from_uri(model_info.model_uri)

>     print(f"serving_example: {serving_example}")

Reviewer 2: Do we want to introduce `validate_serving_input` API here?

> Serve the model

Reviewer 2: ```suggestion
Serve the model locally
```

.. code-block:: bash

    mlflow models serve --model-uri "<YOUR_MODEL_URI>"

> Query the model

Reviewer 2: ```suggestion
Validate model inference with the serving payload example
```
